gamersnet_backend/routes/posts/createPost.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+// include our function from the database to add post
+let {addPost} = require('../../persistence/posts');
+let {getUserByUsername} = require('../../persistence/users')
+
+//to only let existing users to create post, not guests
```

Comment:
This might be in process (as I noticed you stated that user auth is TODO) but this should be reworked to check the cookie token instead of checking client's username parameter

Reply:
I am not very sure how I am going to handle the cookies. Do I look up the token from db by userID? And then compare that with the provided(passed in as a parameter) token/sessionID?

Reply:
So the idea with cookie tokens is that the server logs in from the frontend and gets a token in the backend in the form of a cookie. That cookie gets saved on the browser and can be accessed on any page. Then when the user goes (for example) to create a post, the frontend sends a POST request and the cookie gets automatically sent along to the backend (no need for having any additional parameters). Then the backend will check if 1) the token exists in the cookies and 2) if the token is valid. Checking if the token is valid would involve searching the token collection for any tokens matching the one in the cookie, then if it exists verifying if the token has not expired yet.

I have a dev task to do where I need to change the password, so I can probably finish a working example of token verification soon-ish.

Reply:
Thanks! that explains a lot. I'll try to do it.